### DISCUSSION TITLE
3.11 asyncio will get trio-like Task Groups and Exception Groups

### DISCUSSION THREAD

**Discussion by gnat** [encode/starlette#1741] [Potential Issue]
![FLrcLWAVkAM_VEt](https://user-images.githubusercontent.com/24665/177908384-b30b2eeb-91ff-4413-937e-228c2ecb1c3b.jpeg)

https://mobile.twitter.com/1st1/status/1493748843430567942

https://realpython.com/python311-exception-groups/

https://bugs.python.org/issue46752

What does this mean for Starlette? Is it possible we may be removing the AnyIO integration and returning to zero dependencies for 1.0? #51 

Related: #1157 #1700 #1678

**Comment by adriangb** [comment]
I don't think so, not any time soon at least. For starters 3.11 won't be the minimum supported Python version for like 4 years.

I also don't see how #1700 and #1678 are related to this: they are not cause by anyio, we'd have all of the same problems using the standard librarie's TaskGroup and most of the same problems using `asyncio.create_task`.

**Reply by rednafi** [reply]
Not only this. Correct me if I'm wrong, `anyio` also gives us compatibility with other event loop implementations like `Trio` or `Curio`.

**Reply by adriangb** [reply]
Yes, indeed

**Reply by Kludex** [reply]
Not curio

**Comment by florimondmanca** [comment]
The way I see it, additions to Python like these are primarily for the benefit of end-users. Library authors don't get to use these newest additions unless a/ there are installable backports (eg. `async-generator`) or b/ libraries drop support for previous versions.

Although `asyncio.TaskGroup` relies on [PEP-654](https://peps.python.org/pep-0654/) which (unfortunately?) adds syntax (`except*`), backports might be made available by unwrapping in plain pre-3.11 code the [logic](https://peps.python.org/pep-0654/#handling-exception-groups) that `except*` implements.
